src/main/java/com/vaadin/recipes/recipe/visiblyclicked/VisiblyClicked.java

```diff
@@ -0,0 +1,66 @@
+package com.vaadin.recipes.recipe.visiblyclicked;
+
+import com.vaadin.flow.router.Route;
+import com.vaadin.recipes.recipe.Metadata;
+import com.vaadin.recipes.recipe.Recipe;
+
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.*;
+import com.vaadin.flow.dom.Element;
+
+@Route("visibly-clicked-button")
+@Metadata(
+  howdoI = "Client-side visual proof of server-side activation",
+  description = "Actions that result in programmatic activation of a Vaadin button don't trigger any visual change to that button. This recipe shows how to make this work."
+)
+public class VisiblyClicked extends Recipe {
+
+  public VisiblyClicked() {
+
+    class PButton extends Button {
+        PButton(String arg) {
+            super(arg);
+        }
+
+        // this method takes care of clickInClient()
+        @Override
+        public void clickInClient() {
```

Comment:
This is not used in the recipe, should it be?

Reply:
I left it out for brevity. I could replace the secondary button action to ClickInClient and then it would be used